### ISSUE TITLE
[epic] Allow unused variables

### ISSUE THREAD

**Issue by rbalicki2** [isographlabs/isograph#124]
Low priority... but, in the long run:

* Depends on #123 
* We need some way to allowlist unused variables, e.g. a directive on the variable definition or something. I think that's not allowed by the parsing quite yet, so we need to:
   * add support to the parser for that
   * `#[derive(Deserialize)]` on some struct, probably an empty struct (e.g. `struct AllowedToBeMissingOptions {}`)
   * Add `allowed_to_be_missing: Option<AllowedToBeMissingOptions>` to `VariableDefinition`
   * Modify the work in #123 to allow the variables where `allowed_to_be_missing.is_some()` is true
   * Skip variables that are entirely unused from being included in the generated query
